.eslintrc

```diff
@@ -42,6 +43,10 @@
       {
         "selector": "AssignmentExpression[left.property.name='href'][right.type=/(Template)?Literal/]",
         "message": "Do not assign window.location.href to a string or string template to avoid losing i18n parameters"
+      },
+      {
+        "selector": "CallExpression[callee.object.name=/^(it|describe|context)$/][callee.property.name='only']",
+        "message": "Test exclusivity should not be committed"
       }
```

Comment:
This isn't working unfortunately, because we disable the rule in the `overrides` section below:

https://github.com/18F/identity-idp/blob/ba82f8727dc93e368829e23d01c4f4745ab15ad9/.eslintrc#L79

Unfortunately disabling the rule is an "all-or-nothing", and we can't preserve specific selectors we want to allow to apply in our tests.

I'm currently considering a couple options:

- Just use `eslint-plugin-mocha`
- Re-evaluate #4593 . Specifically, I'm having a hard time reproducing the original issue, the referenced issue https://github.com/babel/babel/issues/8449 was closed as a non-issue, and the [Babel REPL example I'd put together at the time](https://babeljs.io/repl#?browsers=ie%2011&build=&builtIns=usage&spec=false&loose=false&code_lz=MYewdgzgLgBA7gJwIYAcUFMERgXhgbQDpiATEYAVwFt0wpCBHCzATwGV0AbdYKEBAIKdOACgDkhCCgCWYMJgC0AIwpQ-YMQEoAugG4AUAHpDMAQmQtCAMwQgqIxKgxZNuoA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.12.12&externalPlugins=) was not problematic in the way I thought it was at the time. Needs more testing.

Reply:
This was quite the rabbit hole.

>Specifically, I'm having a hard time reproducing the original issue

I can reproduce it, but only when compiled for production.

The original error, for reference: "Invalid attempt to spread non-iterable instance. In order to be iterable, non-array objects must have a \[Symbol.iterator\]() method."

>the [Babel REPL example I'd put together at the time](https://babeljs.io/repl#?browsers=ie%2011&build=&builtIns=usage&spec=false&loose=false&code_lz=MYewdgzgLgBA7gJwIYAcUFMERgXhgbQDpiATEYAVwFt0wpCBHCzATwGV0AbdYKEBAIKdOACgDkhCCgCWYMJgC0AIwpQ-YMQEoAugG4AUAHpDMAQmQtCAMwQgqIxKgxZNuoA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.12.12&externalPlugins=) was not problematic in the way I thought it was

The example is problematic. The problem is that the right-side transpiled version contains a direct reference to `Array.from`, despite the fact `Array.from` isn't polyfilled.

---

After thrashing about a bit, I created an isolated example, and discovered that the output becomes fixed in the very-recently-released Babel 7.13.x (specifically `@babel/preset-env@7.13.0`). The REPL on the site hasn't yet been updated, and still runs 7.12.x.

Before / After: https://gist.github.com/aduth/d7d3098eddae30aef14117fe0156a6da

From [the release notes](https://github.com/babel/babel/releases/tag/v7.13.0), nothing in particular sticks out as having been the fix.

tl;dr: We could probably upgrade Babel, revert the changes from #4593 (or at least remove the ESLint restriction), and safely use array spread if we'd want to. I confirmed this works locally in Internet Explorer after upgrading.

For now, I think it may be reasonable to at least remove those ESLint restrictions, so the override isn't disabling the rule for test exclusivity.